include/swift/AST/DiagnosticsSema.def

```diff
@@ -1504,6 +1504,9 @@ ERROR(access_control_extension_open,none,
 ERROR(access_control_open_bad_decl,none,
       "only classes and overridable class members can be declared 'open';"
       " use 'public'", ())
+ERROR(access_control_non_objc_open_func,none,
+      "non-@objc functions declared inside extensions cannot use 'open' as their default access; use 'public'",
```

Comment:
This is potentially a good opportunity to briefly explain why `open` cannot be used. Say something like:

```
"non-@objc %select{methods|properties}0 in extensions cannot be overriden; use 'public' instead"
```